src/append_only_data.rs

```diff
@@ -663,6 +663,28 @@ where
     P: Permissions + std::hash::Hash + Clone,
 {
     fn append(&mut self, entries: &[(Vec<u8>, Vec<u8>)], last_entries_index: u64) -> Result<()> {
```

Comment:
I think it may be better to just pass a `Vec` here rather than a slice, since the collection will be consumed by this function.  I.e. changing the signature to `mut entries: Entries`.